<!DOCTYPE html>
<html lang="en">

  <head>

    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
    <meta name="description" content="Bioinformatics Docker Images Project: Docker images for common bioinformatics software">
    <meta name="author" content="Instituto de Biología Molecular y Celular (IBMC)">

    <title>Bioinformatics Docker Images Project</title>

    <!-- Bootstrap core CSS -->
    <link href="vendor/bootstrap/css/bootstrap.min.css" rel="stylesheet">

    <!-- Custom styles for this template -->
    <link href="css/scrolling-nav.css" rel="stylesheet">

    <!-- Global site tag (gtag.js) - Google Analytics -->
    <script async src="https://www.googletagmanager.com/gtag/js?id=UA-133688766-1"></script>
    <script>
      window.dataLayer = window.dataLayer || [];
      function gtag(){dataLayer.push(arguments);}
      gtag('js', new Date());

      gtag('config', 'UA-133688766-1');
    </script>
  </head>

  <body id="page-top">

    <!-- Navigation -->
    <nav class="navbar navbar-expand-lg navbar-dark bg-dark fixed-top" id="mainNav">
      <div class="container">
        <a class="navbar-brand js-scroll-trigger" href="#page-top">Home</a>
        <button class="navbar-toggler" type="button" data-toggle="collapse" data-target="#navbarResponsive" aria-controls="navbarResponsive" aria-expanded="false" aria-label="Toggle navigation">
          <span class="navbar-toggler-icon"></span>
        </button>
        <div class="collapse navbar-collapse" id="navbarResponsive">
          <ul class="navbar-nav ml-auto">
            <li class="nav-item">
              <a class="nav-link js-scroll-trigger" href="#about">About</a>
            </li>
            <li class="nav-item">
              <a class="nav-link js-scroll-trigger" href="#install">Install Docker</a>
            </li>
            <li class="nav-item">
              <a class="nav-link js-scroll-trigger" href="#images">Images</a>
            </li>
            <li class="nav-item">
              <a class="nav-link js-scroll-trigger" href="#contact">Contact</a>
            </li>
            <li>
              <a href="https://github.com/pegi3s/dockerfiles"><img src="images/GitHub-Mark-Light-32px.png" alt="GitHub" class="header-img"></a>
            </li>
          </ul>
        </div>
      </div>
    </nav>

    <header class="bg-primary text-white">
     <div class="container text-center">
        <h1>Bioinformatics Docker Images Project</h1>
        <p class="lead">Phenotypic Evolution Group - IBMC/i3S</p>
        <a class="navbar-brand js-scroll-trigger" href="https://hub.docker.com/u/pegi3s/"><img src="images/pegi3s_logo.png" alt="pegi3s"></a>
     </div>
    </header>

    <section id="about">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2>About this page</h2>
            <p class="lead">This website contains information about the Docker images maintained by the <a href="https://www.i3s.up.pt/research-group?x=44">Phenotypic Evolution Group (pegi3s)</a> at the <a href="https://www.ibmc.up.pt/">Instituto de Biologia Molecular e Celular (IBMC)</a> / <a href="https://www.i3s.up.pt/">Instituto de Investigação e Inovação em Saúde da Universidade do Porto (i3S)</a>. All images are available through our <a href="https://hub.docker.com/u/pegi3s/">Docker Hub repository</a>. 
            
            <br></br>
            
            In addition, this repository also contains tutorials to serve as guidance on the usage of existing Docker images (e.g. Trinity for RNA-Seq).</p>
          </div>
        </div>
      </div>
    </section>

    <section id="advantages">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2>Docker Images Advantages</h2>
            <ul>
            <li>Easy to use since it only requires the installation of Docker.</li>
            <li>Portable between computers and immutable.</li>
            <li>Can be deposited in public databases such as <b>Docker Hub</b>.</li>
            <li>Can be downloaded when needed and erased when no longer needed.</li>
            <li>The installed scientific software is ready to run.</li>
            <li>Software invoked through the command line can be directly run in <b>Linux</b> and <b>Windows</b> platforms.</li>
            <li>Docker images can be used in pipelines.</li>
            </ul>
          </div>
        </div>
      </div>
    </section>

    <section id="install">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2>Install Docker</h2>
            <p class="lead">In order to use the Docker images maintained by <a href="https://www.i3s.up.pt/research-group?x=44">pegi3s</a>, users need to install Docker first.</p>
            <h4>Linux</h4>
            <p class="lead">To install Docker in Linux, please follow the steps provided <a href="https://www.digitalocean.com/community/tutorials/how-to-install-and-use-docker-on-ubuntu-16-04">here</a> (additional information can be found <a href="https://docs.docker.com/install/linux/docker-ce/ubuntu/">here</a>).</p>
            <h4>Windows</h4>
            <p class="lead">To install Docker in Windows, please download the official installer <a href="https://download.docker.com/win/stable/Docker%20for%20Windows%20Installer.exe">Docker for Windows Installer.exe</a>. If your system does not meet the requirements to run Docker for Windows (available <a href="https://docs.docker.com/docker-for-windows/install/#what-to-know-before-you-install">here</a>), please install <a href="https://docs.docker.com/toolbox/overview/">Docker Toolbox</a> instead.</p>
            <h4>VirtualBox</h4>
            <p class="lead">As an alternative and for practical purposes, we provide an Ubuntu 18.04 LTS (Linux) virtual image with 11.8 GB, with Docker and <a href="https://www.portainer.io/">Portainer</a> already installed. Please note that this virtual image is limited to 100 GB. In order to run it, first download VirtualBox <a href="https://www.virtualbox.org/">here</a>. After installing VirtualBox, users just need to import the OVA file <a href="https://drive.google.com/file/d/1A2ZPILnWe2QOeCrcBBDcowlrsPsrDHti/view?usp=sharing">here</a> provided. To import an existing virtual machine into VirtualBox, please follow the steps described <a href="https://docs.oracle.com/cd/E26217_01/E26796/html/qs-import-vm.html">here</a>.The administrator's user and password is <b>pegi3s</b>. The username and password for <a href="https://www.portainer.io/">Portainer</a> is <b>pegi3sdocker</b>.</p>
          </div>
        </div>
       </div>
     </section>

    <section id="images">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2>Images</h2>
            <p class="lead">List of available bioinformatic images:</p>
            <ul>
            <li><a href="https://hub.docker.com/r/pegi3s/abyss/"><b>abyss</b></a> 
            <a href="https://github.com/bcgsc/abyss/blob/master/README.md">[doc]</a> - <i>de novo</i> sequence assembler</li>
            <li><a href="https://hub.docker.com/r/pegi3s/adops-gui/"><b>adops</b></a> 
            <a href="http://www.sing-group.org/ADOPS/">[doc]</a> - Philogenetics and detection of positively selected sites</li>
            <li><a href="https://hub.docker.com/r/pegi3s/afterqc/"><b>afterqc</b></a> 
            <a href="https://github.com/OpenGene/AfterQC/blob/master/README.md">[doc]</a> - Sequence read quality assessment</li>
            <li><a href="https://hub.docker.com/r/pegi3s/alter/"><b>alter</b></a> 
            <a href="http://sing-group.org/ALTER/help/ALTER-UserGuide.htm">[doc]</a> - File conversion</li>            
            <li><a href="https://hub.docker.com/r/pegi3s/autoaugustus/"><b>autoaugustus</b></a> 
            <a href="https://github.com/Gaius-Augustus/Augustus/blob/master/scripts/README.autoAug">[doc]</a> - Gene annotation</li>  
            <li><a href="https://hub.docker.com/r/pegi3s/bdbm/"><b>bdbm</b></a> 
            <a href="https://www.sing-group.org/BDBM/manual.html">[doc]</a> - Sequence database manager</li>                
            <li><a href="https://hub.docker.com/r/pegi3s/bedtools/"><b>bedtools</b></a> 
            <a href="http://bedtools.readthedocs.io/en/latest/content/overview.html">[doc]</a> - General utilities</li>
            <li><a href="https://hub.docker.com/r/pegi3s/blast/"><b>blast</b></a> 
            <a href="https://www.ncbi.nlm.nih.gov/books/NBK279690/">[doc]</a> - Sequence alignment</li>
            <li><a href="https://hub.docker.com/r/pegi3s/clustalomega/"><b>clustalomega</b></a> 
            <a href="https://www.ebi.ac.uk/seqdb/confluence/display/THD/Clustal+Omega">[doc]</a> - Sequence alignment</li>
            <li><a href="https://hub.docker.com/r/pegi3s/coral/"><b>coral</b></a> 
            <a href="https://www.cs.helsinki.fi/u/lmsalmel/coral/">[doc]</a> - Sequence read correction</li>
            <li><a href="https://hub.docker.com/r/pegi3s/cutadapt/"><b>cutadapt</b></a> 
            <a href="http://cutadapt.readthedocs.io/en/stable/guide.html">[doc]</a> - Sequence read trimming</li>
            <li><a href="https://hub.docker.com/r/pegi3s/edena/"><b>edena</b></a> 
            <a href="http://www.genomic.ch/edena.php">[doc]</a> - <i>de novo</i> sequence assembler</li>
            <li><a href="https://hub.docker.com/r/pegi3s/emboss/"><b>emboss</b></a> 
            <a href="http://emboss.sourceforge.net/docs/">[doc]</a> - General utilities</li>
            <li><a href="https://hub.docker.com/r/pegi3s/fastqc/"><b>fastqc</b></a> 
            <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/">[doc]</a> - Sequence read quality assessment</li>
            <li><a href="https://hub.docker.com/r/pegi3s/fasttree/"><b>fasttree</b></a> 
            <a href="http://www.microbesonline.org/fasttree/#How">[doc]</a> - Philogenetics inferences</li>
            <li><a href="https://hub.docker.com/r/pegi3s/flash/"><b>flash</b></a> 
            <a href="http://ccb.jhu.edu/software/FLASH/">[doc]</a> - Sequence read overlap</li>
            <li><a href="https://hub.docker.com/r/pegi3s/gbif_gis/"><b>gbif_gis</b></a> 
            <a href="https://www.gbif.org/what-is-gbif">[doc]</a> - Species distribution software</li>    
            <li><a href="https://hub.docker.com/r/pegi3s/gis/"><b>gis</b></a> 
            <a href="https://en.wikipedia.org/wiki/Geographic_information_system">[doc]</a> - Species distribution software</li>    
            <li><a href="https://hub.docker.com/r/pegi3s/haddock_server/"><b>haddock_server</b></a> 
            <a href="http://www.bonvinlab.org/software/haddock2.2/manual/">[doc]</a> - Protein docking prediction</li>
            <li><a href="https://hub.docker.com/r/pegi3s/hyphy/"><b>hyphy</b></a> 
            <a href="http://hyphy.org/tutorials/CL-prompt-tutorial/">[doc]</a> - Philogenetics inferences</li> 
            <li><a href="https://hub.docker.com/r/pegi3s/jmodeltest2/"><b>jmodeltest2</b></a> 
            <a href="https://usermanual.wiki/Document/manual.230794443.pdf">[doc]</a> - Best-fit models of nucleotide substitution</li>
            <li><a href="https://hub.docker.com/r/pegi3s/mauve/"><b>mauve</b></a> 
            <a href="http://darlinglab.org/mauve/mauve.html">[doc]</a> - Genome alignment</li>
            <li><a href="https://hub.docker.com/r/pegi3s/maxbin2/"><b>maxbin2</b></a> 
            <a href="https://raw.githubusercontent.com/pegi3s/dockerfiles/master/maxbin2/2.2.4/options_menu/options.txt">[doc]</a> - Software for binning assembled metagenomic sequences</li>              
            <li><a href="https://hub.docker.com/r/pegi3s/mrbayes/"><b>mrbayes</b></a> 
            <a href="http://mrbayes.sourceforge.net/mb3.2_manual.pdf">[doc]</a> - Philogenetics inferences</li>
            <li><a href="https://hub.docker.com/r/pegi3s/omegamap/"><b>omegamap</b></a> 
            <a href="https://github.com/danny-wilson/omegaMap/blob/master/Documentation/documentation.pdf">[doc]</a> - Detection of positively selected sites</li>
            <li><a href="https://hub.docker.com/r/pegi3s/paml/"><b>paml</b></a> 
            <a href="http://abacus.gene.ucl.ac.uk/software/pamlDOC.pdf">[doc]</a> - Detection of positively selected sites</li>
            <li><a href="https://hub.docker.com/r/pegi3s/phipack/"><b>phipack</b></a> 
            <a href="https://www.maths.otago.ac.nz/~dbryant/software/phimanual.pdf">[doc]</a> - Recombination tests</li>   
            <li><a href="https://hub.docker.com/r/pegi3s/plasflow/"><b>plasflow</b></a> 
            <a href="https://github.com/smaegol/PlasFlow/blob/master/README.md">[doc]</a> - Plasmid prediction</li> 
            <li><a href="https://hub.docker.com/r/pegi3s/popart/"><b>popart</b></a> 
            <a href="http://popart.otago.ac.nz/documentation.shtml">[doc]</a> - Population genetics software</li>
            <li><a href="https://hub.docker.com/r/pegi3s/pss-fs/"><b>pss-fs</b></a> 
            <a href="https://github.com/pegi3s/pss-fs">[doc]</a> - Detection of positively selected sites</li>
            <li><a href="https://hub.docker.com/r/pegi3s/raxml/"><b>raxml</b></a> 
            <a href="https://sco.h-its.org/exelixis/resource/download/NewManual.pdf">[doc]</a> - Philogenetics inferences</li>    
            <li><a href="https://hub.docker.com/r/pegi3s/sapp/"><b>sapp</b></a> 
            <a href="http://sapp.gitlab.io/">[doc]</a> - Genomic tools</li>
            <li><a href="https://hub.docker.com/r/pegi3s/seda/"><b>seda</b></a> 
            <a href="http://www.sing-group.org/seda/manual/index.html">[doc]</a> - Sequence dataset builder</li>
            <li><a href="https://hub.docker.com/r/pegi3s/seqkit/"><b>seqkit</b></a> 
            <a href="https://bioinf.shenwei.me/seqkit/usage/">[doc]</a> - FASTA/Q toolkit</li>
            <li><a href="https://hub.docker.com/r/pegi3s/soapdenovo2/"><b>soapdenovo2</b></a> 
            <a href="https://github.com/aquaskyline/SOAPdenovo2/blob/master/README.md">[doc]</a> - <i>de novo</i> sequence assembler</li>
            <li><a href="https://hub.docker.com/r/pegi3s/spades/"><b>spades</b></a> 
            <a href="http://spades.bioinf.spbau.ru/release3.11.1/manual.html">[doc]</a> - <i>de novo</i> sequence assembler</li>   
            <li><a href="https://hub.docker.com/r/pegi3s/splitstree/"><b>splitstree</b></a> 
            <a href="http://ab.inf.uni-tuebingen.de/data/software/splitstree4/download/manual.pdf">[doc]</a> - Recombination analysis</li>    
            <li><a href="https://hub.docker.com/r/pegi3s/sppider/"><b>sppider</b></a> 
            <a href="http://sppider.cchmc.org/sppider_doc.html">[doc]</a> - Prediction of protein-protein interfaces</li>
            <li><a href="https://hub.docker.com/r/pegi3s/sratoolkit/"><b>sratoolkit</b></a> 
            <a href="https://trace.ncbi.nlm.nih.gov/Traces/sra/sra.cgi?view=toolkit_doc">[doc]</a> - Operations on SRA database</li>
            <li><a href="https://hub.docker.com/r/pegi3s/taxodros_gis/"><b>taxodros_gis</b></a> 
            <a href="https://www.taxodros.uzh.ch/history.php">[doc]</a> - Species distribution software</li>
            <li><a href="https://hub.docker.com/r/pegi3s/tcoffee/"><b>tcoffee</b></a> 
            <a href="https://media.readthedocs.org/pdf/tcoffee/latest/tcoffee.pdf">[doc]</a> - Sequence alignment</li>
            <li><a href="https://hub.docker.com/r/pegi3s/treeviewx/"><b>treeviewx</b></a> 
            <a href="https://code.google.com/archive/p/treeviewx/">[doc]</a> - Displays philogenetics trees</li>
            <li><a href="https://hub.docker.com/r/pegi3s/trinity/"><b>trinity</b></a> 
            <a href="https://github.com/trinityrnaseq/trinityrnaseq/wiki">[doc]</a> - <i>de novo</i> transcript assembly and quantification</li>
            <li><a href="https://hub.docker.com/r/pegi3s/utilities/"><b>utilities</b></a> - General utilities</li>
            <li><a href="https://hub.docker.com/r/pegi3s/velvet/"><b>velvet</b></a> 
            <a href="https://www.ebi.ac.uk/~zerbino/velvet/Manual.pdf">[doc]</a> - Sequence assembler for very short reads</li>
            <li><a href="https://hub.docker.com/r/pegi3s/whiscy/"><b>whiscy</b></a> 
            <a href="https://github.com/haddocking/whiscy">[doc]</a> - Prediction of protein-protein interfaces</li>
            </ul>
            <p class="lead">Additional images:</p>
            <ul>
            <li><a href="https://hub.docker.com/r/pegi3s/formfind/"><b>formfind</b></a> 
            <a href="https://github.com/VR51/formfind">[doc]</a> - HTML processing</li>
            <li><a href="https://hub.docker.com/r/pegi3s/r_project/"><b>r_project</b></a> 
            <a href="https://cran.r-project.org/manuals.html">[doc]</a> - Statistical computing</li>
            </ul>
          </div>
        </div>
      </div>
    </section>

    <section id="remarks">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2>Final Remarks</h2>
            <ul>
            <li>Please note that the original software licenses still apply.</li>
            <li>If there is not a Dockerfile tab on an image repository just add "<b>dockerfile</b>" to the end of the Docker image URL.</li>
            <li><code>sudo su</code> (administrator privileges) is required to run any image. As an alternative please follow the steps described <a href="https://docs.docker.com/install/linux/linux-postinstall/">here</a> to manage Docker as a non-root user.</li>
            <li>To purge all unused or dangling images, containers, volumes and networks, please use the following command:</li>
            <code>docker system prune -a</code>
            <li>To remove one or more specific images, please use instead the following command:</li>                
            <code>docker rmi &lt;image_name&gt; or &lt;image_ID&gt;</code>
            </ul>
          </div>
        </div>
      </div>
    </section>

    <section id="contact">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2>Get in Touch</h2>
            <p class="lead">If you are interested in our project or would like to contribute with suggestions, you may contact us through <a href="&#109;&#097;&#105;&#108;&#116;&#111;:&#112;&#101;&#103;&#105;&#051;&#115;&#100;&#111;&#099;&#107;&#101;&#114;&#064;&#103;&#109;&#097;&#105;&#108;&#046;&#099;&#111;&#109;">&#112;&#101;&#103;&#105;&#051;&#115;&#100;&#111;&#099;&#107;&#101;&#114;&#064;&#103;&#109;&#097;&#105;&#108;&#046;&#099;&#111;&#109;</a>.</p>
          </div>
        </div>
      </div>
    </section>

    <section id="Dissemination">
      <div class="container">
        <div class="row">
          <div class="col-lg-8 mx-auto">
            <h2>Dissemination</h2>
            <p class="lead">P. Ferreira; H. López-Fernández; P. Duque; Cristina P. Vieira; F. Fdez-Riverola; Miguel Reboiro-Jato; J. Vieira (2019) <a href="https://doi.org/10.5281/zenodo.2573107">Bioinformatics Docker Images</a>. Bioinformatics Open Days: BOD 2019. Braga, Portugal. 21 - February <a href="https://doi.org/10.5281/zenodo.2573107"><img src="https://zenodo.org/badge/DOI/10.5281/zenodo.2573107.svg" alt="DOI"></a></p>
          </div>
        </div>
      </div>
    </section>

    <section id="logos">
        <div class="container text-center">
            <div class="row">
                <div class="col-lg-8 mx-auto">
                    <a class="navbar-brand js-scroll-trigger" href="https://biodata.pt/"><img src="images/logo_biodata.png" alt="logo_biodata" class="footer-logo"></a>
                    <br></br>
                    <p align="left" class="footer-logo">Co-funded by:</p>
                    <a><img src="images/apoios_biodata.png" alt="apoios_biodata" class="footer-logo"></a>
                </div>
            </div>
        </div>
    </section>

    <!-- Footer -->
    <footer class="py-5 bg-dark">
      <div class="container">
        <p class="m-0 text-center text-white">Copyright &copy; Phenotypic Evolution Group 2019</p>
      </div>
      <!-- /.container -->
    </footer>

    <!-- Bootstrap core JavaScript -->
    <script src="vendor/jquery/jquery.min.js"></script>
    <script src="vendor/bootstrap/js/bootstrap.bundle.min.js"></script>

    <!-- Plugin JavaScript -->
    <script src="vendor/jquery-easing/jquery.easing.min.js"></script>

    <!-- Custom JavaScript for this theme -->
    <script src="js/scrolling-nav.js"></script>

  </body>

</html>
